core/store.go

```diff
@@ -238,6 +240,37 @@ func Rename(sourceKey, destKey string) bool {
 	}, WithStoreLock(), WithKeypoolLock())
 }
 
+// TODO, Key expiry should be handled for the RandomKey command
+// with the below implementations key expiry checks will be difficult(in Redis expiration is checked with maxTries of 100)
+// this should be considered on future data structure refactoring
+func RandomKey() string {
+	randomKey := ""
+	withLocks(func() {
+		if len(keypool) == 0 {
+			return
+		}
+		idxRange := len(keypool) / 2
+		if idxRange > 1024 {
+			idxRange = 1024
+		}
+
+		index, err := rand.Int(rand.Reader, big.NewInt(int64(idxRange)))
+		if err != nil {
+			return
+		}
+		RandomIndex := int(index.Int64())
+		currentIndex := 0
+		for key := range keypool {
+			if currentIndex == RandomIndex {
+				randomKey = key
+				return
+			}
+			currentIndex++
+		}
+	}, WithStoreLock(), WithKeypoolLock())
```

Comment:
1. Why are we using a write lock on store here? Can we remove it if not necessary?
2. Could we use a read lock (RLock) for keypool instead of a write lock?

Reply:
@JyotinderSingh whats your thought on this ?

Reply:
This makes sense. No need to lock the store, and take a read lock on the keypool